shards/sched.go

```diff
@@ -56,17 +56,26 @@ import (
 // allow improvements as we learn more.
 type scheduler struct {
 	// Exclusive process holds a write lock, search processes hold read locks.
-	mu       sync.RWMutex
-	throttle *semaphore.Weighted
+	mu             sync.RWMutex
+	semInteractive *semaphore.Weighted
+	semBatch       *semaphore.Weighted
 
 	// interactiveDuration is how long we run a search query at interactive
 	// priority before downgrading it to a batch/slow query.
 	interactiveDuration time.Duration
 }
 
 func newScheduler(capacity int64) *scheduler {
+	// Burst upto 1/4 of CPUs for batch. This means we now can use upto 1.25 the
+	// amount of CPU.
+	batchCap := capacity / 4
```

Comment:
> I wonder if we run into trouble with resource limits on Kubernetes. Can we be certain that this doesn't cause a crash loop? Do we have to update admin documentation?

This can lead to use consuming more memory. But the hope is with streaming that we in total have drastically reduced our memory usage so overall we will be fine. Both IO and CPU will be more contested though. As we discussed yesterday I will be updating this PR to add "hidden" tuneables for this.

> Edit: Along the same lines: is it really correct to talk about CPUs here? If I understand correctly, we allow 1 search per CPU but we don't really know how that translates to actuall load, do we?

You are correct, the comment is misleading. I will update.